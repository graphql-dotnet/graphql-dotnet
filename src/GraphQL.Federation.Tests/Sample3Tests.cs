using GraphQL.Federation.CodeFirst.Sample3;
using Microsoft.AspNetCore.Mvc.Testing;

namespace GraphQL.Federation.Tests;

public class Sample3Tests
{
    [Fact]
    public async Task Schema()
    {
        await using var webApp = new WebApplicationFactory<Program>();
        var ret = await webApp.Server.ExecuteGraphQLRequest("/graphql", "{ _service { sdl } }");
        ret.GetSdl().ShouldMatchApproved(o => o.NoDiff());
    }

    [Fact]
    public async Task Entities()
    {
        await using var webApp = new WebApplicationFactory<Program>();
        var query = "query ($arg: [_Any!]!) { _entities(representations: $arg) { __typename ... on Product { id reviews { ...reviewFragment } } ... on User { id reviews { ...reviewFragment } } } } fragment reviewFragment on Review { id content product { id } author { id } }";
        var variables = """{ "arg": [{ "__typename": "Product", "id": "1" },{ "__typename": "User", "id": "1" }] }""";
        var cat = await webApp.Server.ExecuteGraphQLRequest("/graphql", query, variables);
        cat.ShouldMatchApproved(o => o.NoDiff());
    }
}
