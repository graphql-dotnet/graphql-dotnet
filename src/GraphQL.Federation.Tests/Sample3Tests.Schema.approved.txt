schema @link(url: "https://specs.apollo.dev/link/v1.0", import: ["@link"]) @link(url: "https://specs.apollo.dev/federation/v2.3", import: ["@key", "@external", "@requires", "@provides", "@shareable", "@inaccessible", "@override", "@tag"]) {
  query: Query
}

scalar _Any

type Query {
  _service: _Service!
  _entities(representations: [_Any!]!): [_Entity]!
}

type _Service {
  sdl: String
}

union _Entity = Review | Product | User

type Review @key(fields: "id") {
  id: ID!
  content: String!
  product: Product!
  author: User!
}

type Product @key(fields: "id") {
  id: ID!
  reviews: [Review!]!
}

type User @key(fields: "id") {
  id: ID!
  reviews: [Review!]!
}
